biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -2678,48 +2699,61 @@ public void clearClasspath() {
 	 */
 	static List<String> ignore = new ExtList<String>(BUNDLE_SPECIFIC_HEADERS);
 
+	/**
+	 * Caller must close this JAR
+	 *
+	 * @param profile
+	 * @return a jar with the executable code
+	 * @throws Exception
+	 */
 	public Jar pack(String profile) throws Exception {
-		Collection< ? extends Builder> subBuilders = getSubBuilders();
+		try (ProjectBuilder pb = getBuilder(null)) {
+			Collection< ? extends Builder> subBuilders = pb.getSubBuilders();
 
-		if (subBuilders.size() != 1) {
-			error("Project has multiple bnd files, please select one of the bnd files").header(EXPORT).context(profile);
-			return null;
-		}
-
-		Builder b = subBuilders.iterator().next();
+			if (subBuilders.size() != 1) {
+				error("Project has multiple bnd files, please select one of the bnd files").header(EXPORT)
+						.context(profile);
+				return null;
+			}
 
-		ignore.remove(BUNDLE_SYMBOLICNAME);
-		ignore.remove(BUNDLE_VERSION);
-		ignore.add(SERVICE_COMPONENT);
+			Builder b = subBuilders.iterator().next();
+
+			ignore.remove(BUNDLE_SYMBOLICNAME);
+			ignore.remove(BUNDLE_VERSION);
+			ignore.add(SERVICE_COMPONENT);
+
+			try (ProjectLauncher launcher = getProjectLauncher();) {
```

Comment:
Minor nit (feel free to ignore), the `;` is extraneous. Seen in a few places where you've converted to try-with-resources.